### ISSUE TITLE
Array index out of bounds in rtrim function

### ISSUE THREAD

**Issue by CPUBarbecue** [japinli/pg_node2graph#1]
I use the command:
`./pg_node2graph -c -n node_color.map nodes/example1.node`
Error:
```
std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::const_reference std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::operator[](size_type) const [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>; const_reference = const char&; size_type = long unsigned int]: Assertion '__pos <= size()' failed.
```
I found that in the rtrim function, the variable `len` may become 0, and `str[len - 1]` will exceed the array range.
```
static string
rtrim(const string& str)
{
	size_t len = str.size();

	while (isspace(str[len - 1])) {
		len--;
	}

	return str.substr(0, len);
}
```
Thank you very much for your tool, it is very helpful to me.

**Comment by japinli** [comment]
@CPUBarbecue Thank you for reporting this issue.  Could you share your `example1.node`?
It can be fixed using the following code:

``` diff
diff --git a/pg_node2graph.cc b/pg_node2graph.cc
index 77f9f27..3c4b182 100644
--- a/pg_node2graph.cc
+++ b/pg_node2graph.cc
@@ -364,7 +364,7 @@ rtrim(const string& str)
 {
        size_t len = str.size();

-       while (isspace(str[len - 1])) {
+       while (len > 0 && isspace(str[len - 1])) {
                len--;
        }

```

**Comment by CPUBarbecue** [comment]
@japinli I am using the example in the project.
https://github.com/japinli/pg_node2graph/blob/main/nodes/example1.node
I modified it like you did and it worked correctly.

**Comment by japinli** [comment]
What platform do you use? I did not encounter this.

**Comment by CPUBarbecue** [comment]
I am using Manjaro Linux.
I tested using the `node_color.map`  in the project. I think my problem was caused by the empty lines in that file.

**Comment by japinli** [comment]
An empty line in `node_color.map` will be ignored.

**Comment by CPUBarbecue** [comment]
In the `load_color_map` function, the judgment of empty lines and comments is after calling the `trim` function.

**Comment by japinli** [comment]
@CPUBarbecue Thanks for your report.  I fixed it as above, see cad2ec77b.  However, I cannot reproduce it on my dev.
